app/models/pglogical_subscription.rb

```diff
@@ -101,15 +100,24 @@ def self.provider_node_attributes(node_name)
 
   private
 
-  def node_name
-    MiqPglogical::NODE_PREFIX + MiqRegion.my_region_number.to_s
+  def update_subscription
+    disable
+    provider_node_name = MiqPglogical.region_to_node_name(provider_region)
+    pglogical.node_dsn_update(provider_node_name, dsn)
+    enable
+  end
+
+  def create_subscription
+    create_node.check if !pglogical.enabled? || self.class.count == 0
```

Comment:
What do you think about adding this fix to the pglogical extension as an `unless_exists` parameter on `#node_create` in a follow up PR?
So when you call `pglogical.node_create("node_name", dsn, true)` you won't get a failure if the node with "node_name" already exists with the same dsn.

Then we can do some more complex logic there to fail if you tried to create a node with a matching name, but a different dsn.